File: go/1155.go
package main

func numRollsToTarget(n int, k int, target int) int {
	if target < n || target > n*k {
		return 0
	}
	const MOD = 1_000_000_007
	dp := make([][]int, n+1)
	for i := range dp {
		dp[i] = make([]int, target+1)
	}
	dp[0][0] = 1
	for i := 1; i < n+1; i++ {
		for j := 1; j < target+1; j++ {
			res := 0
			for d := 1; d < k+1; d++ {
				if j >= d {
					res = (res + dp[i-1][j-d]) % MOD
				}
			}
			dp[i][j] = res
		}
	}
	return dp[n][target]
}

func main() {

}

File: python/1155.py
class Solution:
    def numRollsToTarget(self, n: int, k: int, target: int) -> int:
        if not (n<=target<=n*k):
            return 0
        MOD = 10 ** 9 + 7
        dp = [[0] * (target+1) for _ in range(n+1)]
        dp[0][0] = 1
        for i in range(1, n+1):
            for j in range(1, target+1):
                res = 0
                for d in range(1, k+1):
                    if j >= d:
                        res = (res + dp[i-1][j-d]) % MOD
                        # dp[i][j] += dp[i-1][j-d]
                        # dp[i][j] %= MOD
                dp[i][j] = res
        # print(dp)
        return dp[n][target]
    
s = Solution()
x = s.numRollsToTarget(2,6,7)
print(x)